core/src/main/java/org/elasticsearch/action/support/single/shard/SingleShardRequest.java

```diff
@@ -119,7 +119,7 @@ public void readFrom(StreamInput in) throws IOException {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeOptionalStreamable(internalShardId);
-        out.writeString(index);
+        out.writeOptionalString(index);
```

Comment:
can we add a note here why this is optional? the validate request suggests otherwise...